legion/jupyterlab-plugin/jupyter_legion/handlers/cloud.py

```diff
@@ -0,0 +1,252 @@
+#
+#    Copyright 2019 EPAM Systems
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+import functools
+
+from tornado.web import HTTPError
+
+from legion.sdk.clients.edi import EDIConnectionException, IncorrectAuthorizationToken, RemoteEdiClient
+from legion.sdk.clients.training import ModelTrainingClient, ModelTraining
+from legion.sdk.clients.deployment import ModelDeploymentClient, ModelDeployment
+from legion.sdk.clients.vcs import VcsClient, VCSCredential
+
+from .base import BaseLegionHandler
+from .datamodels.cloud import *
+
+LEGION_CLOUD_CREDENTIALS_EDI = 'X-Legion-Cloud-Endpoint'
+LEGION_CLOUD_CREDENTIALS_TOKEN = 'X-Legion-Cloud-Token'
+
+
+def _decorate_handler_for_exception(function):
+    """
+    Wrap API handler to properly handle EDI client exceptions
+    :param function: function to wrap
+    :return: wrapped function
+    """
+    @functools.wraps(function)
+    def wrapper(*args, **kwargs):
+        try:
+            return function(*args, **kwargs)
```

Comment:
How do you handle situation when a token become outdated?

Reply:
I show error to client "Your credentials are not up-to-date, please..."